qiskit_ibm_runtime/utils/estimator_pub_result.py

```diff
@@ -0,0 +1,130 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+========================================================================================
+EstimatorPubResult result classes (:mod:`qiskit_ibm_runtime.utils.estimator_pub_result`)
+========================================================================================
+
+.. autosummary::
+   :toctree: ../stubs/
+"""
+
+from __future__ import annotations
+
+from typing import Sequence, TYPE_CHECKING
+
+from qiskit.primitives.containers import PubResult
+
+if TYPE_CHECKING:
+    from plotly.graph_objects import Figure as PlotlyFigure
+
+
+class EstimatorPubResult(PubResult):
+    """Result of Estimator Pub."""
+
+    def draw_zne(
```

Comment:
What happens when this is estimator data with ZNE turned off? Some possible behaviour:
1. fails gracefully with warning empty plot
2. fails gracefully with actionable error message
3. fails with cryptic AttributeError or something
4. not relevant, because this type is only returned (future work) by the runtime if there was ZNE done

Reply:
Right now, it checks that the right metadata fields are present and if not throws an error (in the `draw_*` functions). I prefer 2 (what is implemented now but will improve the error message, we point people how to use ZNE and this method) or 4 (but would probably make sense to rename the class to ZnePubResult or something).

Reply:
I am okay with 1, 2 and 4. Personally, thinking of a scenario where we want to include more functionality for different types of results, I find 1 and 2 simpler on the user side, in that:
- with 2 there is one class of EstimatorPubResult, some of its methods require specific metadata to be present and will fail (or return an empty plot with a warning) otherwise
- with 4 there are different classes of estimator results with potentially different methods, so as a user I need to go into the documentation to check what methods are supported by the result class that I'm looking at

But 4 may be easier to handle on the dev side, so i can be okay with 4 too

Reply:
Good points @SamFerracin , you have pushed me into camps 1,2.

Reply:
I went with 2 and changed the error message to be more descriptive.